web/src/main/java/org/openmrs/web/servlet/LoginServlet.java

```diff
@@ -59,6 +59,17 @@
 	private Map<String, Date> lockoutDateByIP = new HashMap<String, Date>();
 	
 	/**
+	 * @see javax.servlet.http.HttpServlet#doGet(javax.servlet.http.HttpServletRequest,
+	 *      javax.servlet.http.HttpServletResponse)
+	 */
+	@Override
+	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		if(Context.getAuthenticatedUser() != null) {
```

Comment:
Don't you need to also check if there was a privilege check that failed like you did in the other commit? Otherwise you don't want a user to be taken to the login page if they are already logged in when they failed a privilege check